mvpa2/base/dataset.py

```diff
@@ -795,7 +795,18 @@ def all_equal(x, y):
     try:
         eq = x == y
     except:
-        return False
+        # we can get here, in case of dictionaries with non-simple values
+        # (e.g. arrays)
+        try:
+            if not set(x.keys()) == set(y.keys()):
+                return False
+            x = [x[k] for k in x]
+            y = [y[k] for k in y]
```

Comment:
@yarikoptic Sorry, incomplete message. The PR that actually had this change and got merged has an improved version without this issue.